internal/planner/operation.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package planner
+
+import (
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/core"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+)
+
+const operationNodeKind string = "operationNode"
+
+// operationNode is the top level node for operations with
+// one or more child selections, such as queries or mutations.
+type operationNode struct {
+	documentIterator
+	docMapper
+
+	children []planNode
+	isDone   bool
+}
+
+func (n *operationNode) Spans(spans core.Spans) {
+	for _, child := range n.children {
+		child.Spans(spans)
+	}
+}
+
+func (n *operationNode) Kind() string {
+	return operationNodeKind
+}
+
+func (n *operationNode) Init() error {
+	n.isDone = false
+	for _, child := range n.children {
+		err := child.Init()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *operationNode) Start() error {
+	for _, child := range n.children {
+		err := child.Start()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *operationNode) Close() error {
+	for _, child := range n.children {
+		err := child.Close()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *operationNode) Source() planNode {
+	return nil
+}
+
+func (p *operationNode) Children() []planNode {
+	return p.children
+}
+
+func (n *operationNode) Explain(explainType request.ExplainType) (map[string]any, error) {
+	switch explainType {
+	case request.SimpleExplain:
+		return map[string]any{}, nil
+
+	case request.ExecuteExplain:
+		return map[string]any{}, nil
```

Comment:
question: is it supposed to be empty?

Did you add explain tests that assert this?

Reply:
I'm not super familiar with the planner package. Is this supposed to return the operations that this node executes? It does appear in the existing explain tests correctly.

Reply:
I believe you can omit the `Explain` method entirely, since its an optional interface. I don't know if there is any context that this planNode can add to the existing plan, so simply not including it in the explain system is an option.

Reply:
As @jsimnz pointed out, making a node explainable is optional. Moreover, `MultiNode` types are not / don't need to be explainable but we use them to wrap the children under them so the explain graph builder I believe handles that case. But when/if you make `operationNode` unexplainable don't forget to leave the added line in the `tests/integration/explain.go` file because the debug explain graph should still need it to assert it properly in the testing framework.

An example of this is `parallelNode` has no `Explain` method, yet still shows in the explain graph to wrap the children under it.